sql/src/main/java/org/apache/druid/sql/calcite/run/NativeQueryMaker.java

```diff
@@ -122,6 +125,24 @@ public boolean feature(QueryFeature feature)
         );
       }
     }
+    int numFilters = plannerContext.getPlannerConfig().getMaxNumericInFilters();
+    if (numFilters < 1) {
+      throw new CannotBuildQueryException("maxNumericInFilters must be greater than 0");
```

Comment:
Is it helpful to add a hint using `PlannerContext#setPlanningError` here? I don't see any usages of that in the `NativeQueryMaker` so I might be wrong.

Reply:
@somu-imply `CannotBuildQueryException` doesn't seem right because it's a configuration error, not query planning error.

> Is it helpful to add a hint using `PlannerContext#setPlanningError` here? I don't see any usages of that in the `NativeQueryMaker` so I might be wrong.

```
  /**
   * Sets the planning error in the context that will be shown to the user if the SQL query cannot be translated
   * to a native query. This error is often a hint and thus should be phrased as such. Also, the final plan can
   * be very different from SQL that user has written. So again, the error should be phrased to indicate this gap
   * clearly.
   */
  public void setPlanningError(String formatText, Object... arguments)
```

@LakshSingla Here is the javadoc of `setPlanningError`. This method can be used when there is an exception thrown during query planning that can be useful for users to fix their queries by themselves to avoid the planning error. Since `queryMaker` is called after query planning is done, it seems fine to not use it.

Reply:
Thanks. Yeah, I realized as well that setting the error afterward would be in vain since it doesn't get surfaced as a hint to the user.